### PR TITLE
PXC Operator 1.6.0 Release Notes

### PR DESCRIPTION
**Pull request by fiowro**
[RN-1](https://jira.percona.com/browse/RN-1)

### CODE REVIEW

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+:Date: September 30, 2020
+:Installation: `Installing Percona Kubernetes Operator for Percona XtraDB Cluster <https://www.percona.com/doc/kubernetes-operator-for-pxc/index.html#quickstart-guides>`_
+
+New Features
+================================================================================
+
+* :jirabug:`K8SPXC-429`: Possibility to restore backups to a new Kubernetes-based environment
+* :jirabug:`K8SPXC-368`: Auto update system users by changing the appropriate secret name
+* :jirabug:`K8SPXC-144`: Allow adding ProxySQL configuration options via ConfigMap
```
Comment (delgod) [inline source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:15]:
```suggestion
* :jirabug:`K8SPXC-144`: Allow adding ProxySQL configuration options
```

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+* :jirabug:`K8SPXC-343`: Helm chart officially provided with the Operator
+
+Improvements
+================================================================================
+
+* :jirabug:`K8SPXC-398`: New crVersion key in ``deploy/cr.yaml`` to indicate the API version that the Custom Resource corresponds to
```
Comment (delgod) [inline source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:21]: is upgrade documentation updated?

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+* :jirabug:`K8SPXC-416`: Support of the HA Proxy proxy-protocol
```
Comment (delgod) [inline source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:22]:
```suggestion
* :jirabug:`K8SPXC-416`: Support of the proxy-protocol in HAProxy
```

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+* :jirabug:`K8SPXC-372`: Support new versions of cert-manager by the Operator
```
Comment (delgod) [inline source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:23]:
```suggestion
* :jirabug:`K8SPXC-372`: Support new versions of cert-manager by the Operator (Thanks to user rf_enigm for reporting this issue)
```

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+* :jirabug:`K8SPXC-317`: The possibility to configure the Operator's imagePullPolicy option (Thanks to user imranrazakhan for reporting this issue)
+* :jirabug:`K8SPXC-438`: Cluster name length limit extended to 32 characters to fit the maximum value allowed by Galera
```
Comment (delgod) [inline source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:25]:
```suggestion
* :jirabug:`K8SPXC-438`: Cluster name length limit extended to 32 characters to fit the maximum value allowed by wsrep_cluster_name
```

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+* :jirabug:`K8SPXC-411`: Extend certmanager configuration to add more domains (multiple SAN) to certificate
```
Comment (delgod) [inline source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:26]:
```suggestion
* :jirabug:`K8SPXC-411`: Extend cert-manager configuration to add more domains (multiple SAN) to certificate
```

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+Bugs Fixed
+================================================================================
+
+* :jirabug:`K8SPXC-431`: HAProxy unable to start on OpenShift with the default ``cr.yaml`` file
+* :jirabug:`K8SPXC-408`: Insufficient MAX_USER_CONNECTIONS=10 for ProxySQL monitor user (increased to 100)
+* :jirabug:`K8SPXC-391`: HA Proxy and PMM cannot be enabled at the same time (Thanks to user rf_enigm for reporting this issue)
```
Comment (delgod) [inline source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:33]:
```suggestion
* :jirabug:`K8SPXC-391`: HAProxy and PMM cannot be enabled at the same time (Thanks to user rf_enigm for reporting this issue)
```

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+* :jirabug:`K8SPXC-406`: Second node (XXX-pxc-1) always selected as donor (Thanks to user pservit for reporting this issue)
+* :jirabug:`K8SPXC-390`: Crash on missing HAProxy PodDisruptionBudget
+* :jirabug:`K8SPXC-355`: Counterintuitive YYYY-DD-MM dates in the S3 backup folder names (Thanks to user graham.webcurl for reporting this issue)
+* :jirabug:`K8SPXC-274`: The 1.2.0 -> 1.3.0 -> 1.4.0 upgrade path not working (Thanks to user martin.atroo for reporting this issue)
+* :jirabug:`K8SPXC-450`: The unnecessary ssls annotations added to HA Proxy Pods are potential reasons of the Pod restart if changed
```
Comment (delgod) [inline source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:38]:
```suggestion
* :jirabug:`K8SPXC-450`: The TLS annotations are reasons of the unnecessary HAProxy Pod restarts
```

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+* :jirabug:`K8SPXC-443`: The outdated version service endpoint URL
```
Comment (delgod) [inline source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:39]: @hors is URL changed between releases?

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+* :jirabug:`K8SPXC-435`: MySQL root password visible through ``kubectl logs``
+* :jirabug:`K8SPXC-426`: mysqld recovery logs not logged to file and not available through ``kubectl logs``
+* :jirabug:`K8SPXC-423`: HA Proxy not refreshing IP addresses even when the node gets different address
```
Comment (delgod) [inline source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:42]:
```suggestion
* :jirabug:`K8SPXC-423`: HAProxy not refreshing IP addresses even when the node gets different address
```

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+* :jirabug:`K8SPXC-419`: Percona XtraDB Cluster incremental state transfers not taken into account by readiness/liveness checks
+* :jirabug:`K8SPXC-418`: HA Proxy not routing traffic for 1 donor, 2 joiners
```
Comment (delgod) [inline source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:44]:
```suggestion
* :jirabug:`K8SPXC-418`: HAProxy not routing traffic for 1 donor, 2 joiners
```

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+* :jirabug:`K8SPXC-417`: Certmanager not compatible with Kubernetes versions below v1.15 due to unnecessarily high API version demand
```
Comment (delgod) [inline source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:45]:
```suggestion
* :jirabug:`K8SPXC-417`: Cert-manager not compatible with Kubernetes versions below v1.15 due to unnecessarily high API version demand
```

**Review by delgod — CHANGES_REQUESTED** [review]

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst (hunk 2 above)
Comment (fiowro) [reply source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:21]: now it is :)

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst (hunk 9 above)
Comment (hors) [reply source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:39]: It was discussed and re-worked.

**Review by delgod — CHANGES_REQUESTED** [review]

**Review by hors — CHANGES_REQUESTED** [review]
please add `K8SPXC-462 add readiness probe for HAproxy`

**Review by delgod — CHANGES_REQUESTED** [review]

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+* :jirabug:`K8SPXC-429`: Possibility to restore backups to a new Kubernetes-based environment :ref:`when needed<backups-restore>`
+* :jirabug:`K8SPXC-144`: Allow adding ProxySQL configuration options
+* :jirabug:`K8SPXC-343`: Helm chart officially provided with the Operator
+
+Improvements
+================================================================================
+
+* :jirabug:`K8SPXC-398`: New crVersion key in ``deploy/cr.yaml`` to indicate the API version that the Custom Resource corresponds to (thanks to user mike.saah for contribution)
+* :jirabug:`K8SPXC-372`: Support new versions of cert-manager by the Operator (thanks to user rf_enigm for contribution)
+* :jirabug:`K8SPXC-317`: Possibility to configure the ``imagePullPolicy`` Operator option (thanks to user imranrazakhan for contribution)
```
[the remark this answers is not in the document]
Comment (fiowro) [reply source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:23]: "thanks for reporting" is reasonable for a bug fix; but these 3 issues are improvements, and it looks like we're thanking user for makinug us know abouw our improvement

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+
+* :jirabug:`K8SPXC-416`: :ref:`Support of the proxy-protocol<haproxy-conf-protocol>` in HAProxy
+* :jirabug:`K8SPXC-429`: Possibility to :ref:`restore backups to a new Kubernetes-based environment<backups-restore>`
+* :jirabug:`K8SPXC-144`: Allow :ref:`adding ProxySQL configuration options<proxysql-conf-custom>`
+* :jirabug:`K8SPXC-343`: Helm chart :ref:`officially provided with the Operator`
+
+Improvements
+================================================================================
+
+* :jirabug:`K8SPXC-394`: Support of :ref:`"cluster-wide" mode<install-clusterwide>` for Percona XtraDB Cluster Operator
```
Comment (delgod) [inline source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:21]: I think K8SPXC-394 is a feature, and K8SPXC-144 is an improvement
up to @spron-in

**Review by delgod — CHANGES_REQUESTED** [review]

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst (hunk 14 above)
Comment (fiowro) [reply source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:21]: I'd say yes, it makes sense to change statuses for these two issues.

Comment (spron-in) [reply source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:21]: Yeah, agree.

File: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+================================================================================
+
+:Date: October 7, 2020
+:Installation: `Installing Percona Kubernetes Operator for Percona XtraDB Cluster <https://www.percona.com/doc/kubernetes-operator-for-pxc/index.html#quickstart-guides>`_
+
+New Features
+================================================================================
+
+* :jirabug:`K8SPXC-416`: :ref:`Support of the proxy-protocol<haproxy-conf-protocol>` in HAProxy
+* :jirabug:`K8SPXC-429`: Possibility to :ref:`restore backups to a new Kubernetes-based environment<backups-restore>`
```
Comment (spron-in) [inline source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst:14]: This one is wrong. It is about restoring a backup without CR. @delgod

Comment (fiowro) [reply source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.6.0.rst]: Changed to "A possibility to restore backups to a new Kubernetes-based environment with no existing Percona XtraDB Cluster Custom Resource". I think mentioning "new environment" is valuable from the user's point of view (actual use cases of the feature).

Also K8SPXC-474 was renamed to  "The init container now has the same resource requests as the main container of a correspondent Pod (Thanks to user yann.leenhardt for contribution)" after the consultation with @qJkee

**Review by delgod — DISMISSED** [review]

**Review by delgod — DISMISSED** [review]

**Review by hors — APPROVED** [review]

**Pull request merged**